src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -340,6 +340,14 @@ public static bool IsFunctionPointer(this TypeSymbol type)
             return type.TypeKind == TypeKind.FunctionPointer;
         }
 
+        public static bool IsPointerOrFunctionPointer(this TypeSymbol type)
+            => type.TypeKind switch
```

Comment:
>=> type.TypeKind switch [](start = 12, length = 23)

Consider using regular switch statement. The expression already spans multiple lines, so an expression body doesn't buy us much. Also, several branches share result and the code is clearer and more maintainable when that doesn't require duplication, I believe.